### ISSUE TITLE
Compiling fails to link on FreeBSD 13

### ISSUE THREAD

**Issue by yonas** [viridIT/vSMTP#609]
**Describe the bug**
Compiling fails to link on FreeBSD 13.

**To Reproduce**
Run `cargo build --workspace --release`.

**Expected behavior**
vsmtp should compile successfully and produce a single binary.

**Screenshots**
```
   Compiling vsmtp-auth v1.3.0 (/usr/home/yonas/git/make-cd/files/vsmtp/src/vsmtp/vsmtp-auth)                                                                                                                        
   Compiling vsmtp-common v1.3.0 (/usr/home/yonas/git/make-cd/files/vsmtp/src/vsmtp/vsmtp-common)                                                                                                                    
   Compiling vsmtp-config v1.3.0 (/usr/home/yonas/git/make-cd/files/vsmtp/src/vsmtp/vsmtp-config)                                                                                                                    
   Compiling vsmtp-delivery v1.3.0 (/usr/home/yonas/git/make-cd/files/vsmtp/src/vsmtp/vsmtp-delivery)                                                                                                                
   Compiling vsmtp-rule-engine v1.3.0 (/usr/home/yonas/git/make-cd/files/vsmtp/src/vsmtp/vsmtp-rule-engine)                                                                                                          
   Compiling vsmtp-server v1.3.0 (/usr/home/yonas/git/make-cd/files/vsmtp/src/vsmtp/vsmtp-server)                                                                                                                    
error: linking with `cc` failed: exit status: 1                                                                                                                                                                      
  |                                                                                                                                                                                                                  
  = note: "cc" "-m64" "/tmp/rustcB8OnkS/symbols.o" ...
...
  = note: ld: error: undefined symbol: __start_linkme_MECHANISMS
          >>> referenced by rsasl.7faa59db-cgu.15
          >>>               rsasl-e533dc3a78a8c0ea.rsasl.7faa59db-cgu.15.rcgu.o:(rsasl::registry::registry_static::MECHANISMS::head3fd0280e5b883) in archive /usr/home/yonas/git/make-cd/files/vsmtp/target/debug/deps/librsasl-e533dc3a78a8c0ea.rlib
          
          ld: error: undefined symbol: __stop_linkme_MECHANISMS
          >>> referenced by rsasl.7faa59db-cgu.15
          >>>               rsasl-e533dc3a78a8c0ea.rsasl.7faa59db-cgu.15.rcgu.o:(rsasl::registry::registry_static::MECHANISMS::head3fd0280e5b883) in archive /usr/home/yonas/git/make-cd/files/vsmtp/target/debug/deps/librsasl-e533dc3a78a8c0ea.rlib
          cc: error: linker command failed with exit code 1 (use -v to see invocation)
```
          

error: could not compile `vsmtp` due to previous error

**System (please complete the following information):**
- vSMTP version: [abe19d0](https://github.com/viridIT/vSMTP/commit/abe19d0b1884f2098a22379103ee89eea60c2db8)
- OS: FreeBSD
- OS version: 13.1-RELEASE

**Additional context**
cargo 1.63.0
cyrus-sasl 2.1.28

**Comment by Mathieu-Lala** [comment]
Thank you for your issue.

The error seems to come from `linkme` which is used by `rsasl + static_registry features`. I removed it here https://github.com/viridIT/vSMTP/pull/621. Can you try to build vsmtp using this branch ?

**Comment by yonas** [comment]
@Mathieu-Lala Thanks, it compiles now!

FYI: For anyone following, make sure to create a `vsmtp` user before running vsmtp or it will crash.

**Comment by yonas** [comment]
Closing. Fixed in #621.
